### PR TITLE
MEB-134, MEB-132: Logo fixes, icon updates, useage guidelines

### PR DESCRIPTION
**Pull request by Aerozol**
Logo fixes
- standardised all logos (fixed inconsistent margins etc)
- added missing logos
- exported 'small' logo versions
- added design-files folder

Icon updates
- added all icons, regular and detailed versions
- ico and svg

Useage guidelines
- logo guidelines added to readme
- added documentation/images directory to contain examples

Useage guidelines are related to/will be compiled into MEB-132

### CODE REVIEW

File: README.md
Comment (reosarevok) [inline README.md]:
```suggestion
\*a specific logo may display fine at a smaller size, but other project logos have more detail, and won’t. Smaller sizes make consistent logo display across projects difficult and are to be avoided. However there may be fringe cases where using smaller sizes is appropriate.
```

File: README.md
Comment (reosarevok) [inline README.md]: You already have `alt="stretched logo"` further down, maybe differentiate this somehow? :)

**Review by akshaaatt — APPROVED** [review]
Lovely! Thanks for all the hardwork :)

File: README.md
Comment (MonkeyDo) [inline README.md]:
```suggestion
>*Don't* alter the logo or add effects such as drop shadows, embossing, etc.
```

File: README.md
Comment (MonkeyDo) [inline README.md:55]: A suggestion for one more "don't" which may not be obvious to everybody.
> *Don't* use the logo on a busy or colorful background

As for an example, perhaps a checkered or color gradient background?

File: README.md
Comment (MonkeyDo) [inline README.md]:
```suggestion
Use logo sizings consistently across projects. For instance, *Brainz logo mini* is used as the top left logo across all MetaBrainz projects. When unsure, refer to prior projects and useage.</br>
```

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Looking waaaaaayyyyy better, thanks for the overhaul !
I have a few suggestions with wording below, as well as these two main changes:


The ListenBrainz logo has changed since the redesign, and all of the LB files (illustrator as well as exports) need to be updated to reflect that.
Here is the illustrator file for v2 of the logo: https://ufile.io/eccjm5zg
We should probably keep the old version too for posterity (at least the illustrator files?).


There are also `desktop.ini` files which look like hidden files* Windows uses for previews, and which shouldn't be added to the repository.
I have the same issue on MacOS which generates .DS_Store hidden files.
We have a few options to resolve this: 
1. I can offer to do some git rebase magic for you to remove these unwanted files from the history
2. I can walk you through how to do said rebase magic yourself, if yesterday's git fun left you wanting some more
3. Blasted be the cleanliness of git history! Delete the unwanted files* and commit those changes (we'll have one more commit in the history, but this is the simplest option)

Whichever option we choose, after the deed is done we will make sure to ignore these files in the future [with gitignore](https://git-scm.com/docs/gitignore).
We will create a hidden file of our own named `.gitignore` in which we'll add two lines of text: `*.ini` and `*.DS_Store`, each on its own line.
Git will then ignore any file (including in subfolders) with these extensions.


 .* You'll need to [show hidden files](https://support.microsoft.com/en-us/windows/view-hidden-files-and-folders-in-windows-97fbc472-c603-9d90-91d0-1166d1d9f4b5) in your windows directory

File: README.md
Comment (Aerozol) [reply README.md]: Thanks Reo! I had pretty much *exactly* written what you had, but shortened it! Very happy to have a second eye on it all, thank you.

Question: If I hit 'commit suggestion' on all of these is it making the pull request complicated again/do I have to squash stuff? Or am I overthinking it and can just 'commit suggestion' and not worry about it.

File: README.md
Comment (Aerozol) [reply README.md]: Oops, got lost in the cut and paste. What's the best way for me to change this at this stage in the process/with a pull request under review?

File: README.md
Comment (yvanzo) [reply README.md]: “Commit suggestion” will just add a commit to your pull request, so it is usually safe to click on it. If you want to keep your pull request tidy, you can always reorganize the commits after pulling the branch from GitHub to your local working copy.

File: README.md
Comment (Aerozol) [reply README.md:51]: > The ListenBrainz logo has changed since the redesign, and all of the LB files (illustrator as well as exports) need to be updated to reflect that.
> Here is the illustrator file for v2 of the logo: https://ufile.io/eccjm5zg
> We should probably keep the old version too for posterity (at least the illustrator files?).

Thanks Monkey!

While updating the LB files I've come across an inconsistency I didn't spot before, in regards to using Roboto or Roboto Condensed. In the 'factory' files it's usually Condensed, but based on what we've actually used on the site icons etc it should probably just be Roboto? Or did you want Condensed for some of the narrower formats.

(this will cause most of the 'factory' exported files to get updated again)

File: README.md
Comment (Aerozol) [reply README.md]: Updated! Thanks @yvanzo

File: README.md
Comment (Aerozol) [reply README.md:55]: Added - resolve if you're happy with it :)

**Review by Aerozol — COMMENTED** [review]
@MonkeyDo, how's this for the gitignore file? Copied from the Picard directory

**Comment by Aerozol** [comment]
> The ListenBrainz logo has changed since the redesign, and all of the LB files (illustrator as well as exports) need to be updated to reflect that.
> Here is the illustrator file for v2 of the logo: https://ufile.io/eccjm5zg
> We should probably keep the old version too for posterity (at least the illustrator files?).

Thanks @MonkeyDo!
(Is there a way to reply to your general comment? I think I must be blind)

While updating the LB files I've come across an inconsistency I didn't spot before, in regards to using Roboto or Roboto Condensed. In the 'factory' files it's usually Condensed, but based on what we've actually used on the site icons etc it should probably just be Roboto? Or did you want Condensed for some of the narrower formats.

(this will cause most of the 'factory' exported files to get updated again)

**Comment by Aerozol** [comment]
> There are also `desktop.ini` files which look like hidden files* Windows uses for previews, and which shouldn't be added to the repository.

@MonkeyDo, huh, I always have hidden files visible and they're not showing any pesky .ini files. So not sure I can do option 2.

Anyway, if you're keen I'd be happy to try catch you and go through how to get rid of them with rebase. Then I'll know exactly what annoyance I'm avoiding/pawning off onto others in future!

**Comment by MonkeyDo** [comment]
Th gitignore file looks perfect, thanks !

For removing the files from the history, here's what I did locally (I had to try it out myself and google some things because, y'know, rebasing is *hard*).
I'll go through the process one thought at a time:

1. First, for simplicity's sake, let's make sure the directory is clean and there are no changes you need to commit.

2. We need to find in which commit the unwanted files were added.
For that, you can use the github website (open each commit [in this PR's "commits" tab](https://github.com/metabrainz/metabrainz-logos/pull/15/commits)) or using Github Desktop (the "history" tab shows the commits) and check if that's where the files were added.
Once I found the commit I'm looking for, I copy its hash somewhere (the long string of characters that identifies a commit, sometimes shortened to the first 7 characters), we'll need it later: 30b834b6a804718536e172769c41daa41cc988ee
On the GH website: image
On Github Desktop: 
image

3. Removing the files
So I'm not sure how you would do this manually if you can't see the hidden files on your machine, but here's a good and simple way to do it in git bash:
`cd` into the metabrainz-logo folder and run this command `git rm **\/desktop.ini`
This translates to "remove all files in all subfolders called desktop.ini"
Not only does that remove the files, but it also stages those changes, ready to be commited.
I run a quick `git status` command to check:
image
I see the files I removes, and it's in green which tells me the changes are staged (ready to be commited)
If yours is also in french, there's probably a problem :D

4. Let's commit those changes
Here we can commit our changes with the 'fixup' flag, which explicitly marks this new commit as a fixup of a specific commit. Here's where we'll be using the commit hash from earlier:
The command to run is: `git commit --fixup 30b834b`
As you see we only need the first 7 characters of the commit hash, but you could also pass the whole hash if in doubt.

I like to do that because later on when we rebase, it will jump out as the fixup commit we're looking for

5. Quick sanity check
Yep, that created a commit with "fixup" and the message of the commit we're fixing:
image

6. Rebasing
For this, we need the hash of the commit right before the one we're fixing up. This will allow us to rebase our history *from that point onwards*.
It's this one: `3be4b4b`
image

With this, we can run the following command to start a rebase in interactive mode (`-i`) and automatically squash our new fixup commit and the commit we're fixing up (30b834b). The 'autosquash' flasg only applies to 'fixup' commits and will squash your fixup commit and the commit it fixes.
`git rebase -i --autosquash 3be4b4b`

Since it's 'interactive' you'll be show a preview of what's about to happen:
As you can see, there is our fixup commit (I know 'cause it's got "fixup!…" in the message) with the 'fixup' command already set for us, and already placed right after the commit we're fixing (30b834b). Everything looks good !
image

7. Finish the rebase by closing the file (if you're in vim, good luck !), and then you can force-push your commits.
We're force-pushing instead of a normal push because we rewrote the history.
Github Desktop doesn't allow you to force-push easily in this case, so you'll have to run the command: `git push --force`

**Comment by Aerozol** [comment]
Hey @MonkeyDo, followed the instructions, amazing thank you! But I see a bunch of commits were added and a merge branch 'master' commit! Uh oh! Guessing that shouldn't have happened? At this stage I'm happy for you to fix everything, I've probably had my git bash fun 😭😛

Let me know re. the regular font vs condensed, there's a little bit of re-work I'll do there once I know.

**Comment by Aerozol** [comment]
Hey @MonkeyDo, I made all fonts 'Roboto', except for the vertical formats, which I left as 'Roboto Condensed'.

I also exported the rest of the formats in the factory files, and added a square format (which Rob had mentioned was missing). 

Hoping that this covers **all** the logo bases for the near future :P

Just one commit, which is nice, but I see some 'remove .ini' edits are in there, which doesn't make sense to me. I'll leave that to the pros from here on in!

Anything else, apart from tidying the commits if needed, before we're done?

**Comment by MonkeyDo** [comment]
Good morning in the future @Aerozol !
I cleanup up the git history of the logos PR, it's now nice and tidy :)
FYI, I used a tool called SourceTree to help me visualize the history, you might find it helpful in the future.

Since I did a rebase to rewrite the git history, and consequently needed to force-push, if from this point onwards you need to modify anything you'll need to nuke your local copy that has the old history. They cannot be reconciled.
To do that, you can run these commands in git bash (I don't know of a way to do it in github desktop unfortunately):
`git fetch origin` will make you local git aware of all that's happened on the repository (AKA the "remote")
`git reset --hard origin/master` will reset you local copy to what is in the remote ("nuke my changes, pull the remote branch 'master' from the repository)


I do have a doubt regarding the new logo files; I don't mind the Roboto instead of Roboto Condensed, it's a lighter look I quite like.
However I'm not a fan of the uppercase vs lowercase size. I must have done some manual changes initially to get taller caps, I don't think they're standard in this font.
Here's a side-by-side comparison (left is after [small caps], right is before [tall caps]); what do you think?
Am I focusing on small details I'll be the only one to see? Even as I write this and look at the side-by-side I must say it makes very little difference. 
image

**Comment by reosarevok** [comment]
I seem to remember the logos were amended at some point because it felt weird that the i dots were higher than the capital letters.

**Comment by Aerozol** [comment]
Morning past @MonkeyDo!
Thanks **so** much, I have updated my local repo.

Logo: Good spotting! I followed the convention in the 'design-factory' files, which is a smaller font size and small caps on the upper case letters. Now I can see that was updated at some point - but usage is mixed. e.g. the ListenBrainz site still has the smaller caps (and RobotoCondensed, so presumably an older version?)

I'll make all the caps match the current MB logo, since both of you note that as a preference.
And _no_ you are not focussing on details that only you'll see :P

re. lighter look that's something I missed: the newer versions are actually 'Roboto Black', not 'Roboto Bold'. I'll update that as well.

**Comment by Aerozol** [comment]
I might leave the smaller caps on the vertically stacked logos:
image

**Comment by atj** [comment]
\<unsolicited opinion\>
The heavier weight (condensed?) font looks better IMO. I think this is primarily because there is too much whitespace between the letters in the new version. Can you reduce the kerning and see how it looks?
\</unsolicited opinion\>

**Comment by Aerozol** [comment]
@atj, thanks, input welcome!
Good spotting, I believe this is because the logos were later updated to use 'Roboto Black' instead of 'Roboto Bold'. I'll update that too. (to clarify, this means all logos will look like MonkeyDo's MusicBrainz example above, the one on the right)
I am thinking I'll leave the vertical stack logos as-is in this respect as well.

Note: Neither of Monkey's examples are condensed - you can see what condensed looks like in the vertical stack above. Squishes everything together!

**Comment by atj** [comment]
The left hand version of the vertically stacked logo actually looks better IMO. However I think it's pretty much impossible to make vertically stacked logos with more than one word/line of text look particularly good, so ¯\\_(ツ)\_/

**Comment by Aerozol** [comment]
@atj omg, sorry that this has gotten so confusing: yup agreed, I want to keep the _left_ hand version on the vertical stack, standardise to the _right_ hand version on the horizontal :P

**Comment by MonkeyDo** [comment]
@Aerozol We're in agreement over these various details: Tall caps (for horizontal logos) and using Roboto Black.

This PR was direly needed, thanks again for sorting all this out, it's such a mess!

I hadn't even realized the ListenBrainz logo on the LB website was wrong… And chances are that I'm the one who made it!

Anyway, great job :)

**Comment by Aerozol** [comment]
Sorry all for this next step taking so long! Work was crazy.

@MonkeyDo, further changes:
- main logos all 77pt font
- all non vertical-stack logos normal caps and Roboto Black font

I've forgotten all the cleanup stuff we went throuhg (of course) but hopefully with just the single commit and our .gitignore file there's nothing to tidy? (optimistic eh)

**Comment by Aerozol** [comment]
Related: https://tickets.metabrainz.org/browse/MEB-135
I've added pull requests to update project sites  that used the old version (BB,P,LB,AB)

File: README.md
Comment (phw) [inline README.md]: Not sure if it is relevant, but on desktop a typical icon size is 32px, and for Picard we start using this icon at that size.

We originally had been using the no detail icon there, but the one with more detail was nicer in many circumstances. See https://github.com/metabrainz/picard/commit/660d5948b07ecfaf797fed7f1c263ac471a6cd05

Comment (Aerozol) [reply README.md]: Wow, pretty close though! Thanks, I've dropped the minimum to 32px to reflect what was tested and worked. I left the print (mm) minimum the same.

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Thank you for the the awesome work, both on this and the guidelines!
Much wow, so clean !

I renamed a couple of files that were misnamed, and something is going on with the messybrainz and listenbrainz icons (in the icons folder), both the .svg and .ico files:
image
image

It also looks like the LB icon is based on the old logo.

apart from that everything is 👄👌  chef's kiss

**Comment by Aerozol** [comment]
Thanks Monkey, great spotting! I found the problem and fixed it (svg doesn't like masks)

Now that there's some diverging branches I'm going to stay away from git until this is closed and I can merge with upstream... Could you update the relevant files? Files here:
https://drive.google.com/drive/folders/1bEbidXz0aZU8lr5uwlU722AByYPa1hlk?usp=sharing

**Comment by MonkeyDo** [comment]
> Now that there's some diverging branches I'm going to stay away from git until this is closed and I can merge with upstream... 

No branching this time, just me adding a commit. For the future, in this situation you should just be able to *pull* the commits to your local repo and then continue as usual.

That being said, I'm on it, will replace the icons!

**Review by MonkeyDo — APPROVED** [review]
That was quite the journey, well done fixing everything!
So much was wrong, and it makes me so happy to see it all so clean.
A million thank yous!

**Pull request merged**
